lab-dustin/test/hero-router-test.js

```diff
@@ -0,0 +1,155 @@
+      .send(data)
+      .then(res => {
+        expect(res.status).toEqual(200);
+        expect(res.body._id).toExist();
+        expect(res.body.name).toExist();
+        expect(res.body.origin).toEqual('A Jar');
+        expect(res.body.goodOrEvil).toEqual('Evil');
+      });
+    });
+    it('should respond with a 400', () => {
+      return superagent.post(`${API_URL}/api/heros`)
+      .catch(res => {
+        // expect(res.status).toEqual(200);
+        expect(res.status).toEqual(400);
+        // expect(res.body.name).toExist();
+        // expect(res.body.origin).toEqual('A Jar');
+        // expect(res.body.goodOrEvil).toEqual('Evil');
+      });
+    });
+    it('should respond with a 409', () => {
+      return superagent.post(`${API_URL}/api/heros`)
+      .send(data).catch(res => {
+        // expect(res.status).toEqual(200);
+        expect(res.status).toEqual(409);
+        // expect(res.body.name).toExist();
+        // expect(res.body.origin).toEqual('A Jar');
+        // expect(res.body.goodOrEvil).toEqual('Evil');
+      });
+    });
+  });
+
+  describe('testing PUT /api/heros/:id', () => {
+    afterEach(() => Hero.remove({}));
+    beforeEach(() => {
+      return new Hero({
+        name: 'Eyeball Man',
+        origin: 'A Head',
+        goodOrEvil: 'Good',
+      })
+      .save()
+      .then(hero => {
+        tempHero = hero;
+      });
+    });
+    it('should respond with a hero', () => {
+      return superagent.put(`${API_URL}/api/heros/${tempHero._id}`)
+      .send({origin: 'A Hand',
+        name: 'Fingerman',
+        goodOrEvil: 'Evil'})
+      .then(res => {
+        expect(res.status).toEqual(200);
+        expect(res.body._id).toEqual(tempHero._id);
+        expect(res.body.name).toEqual('Fingerman');
+        expect(res.body.origin).toEqual('A Hand');
+        expect(res.body.goodOrEvil).toEqual('Evil');
+        expect(new Date(res.body.dateCreated)).toEqual(tempHero.dateCreated);
+      });
+    });
+    it('should respond with 400', () => {
+      return superagent.put(`${API_URL}/api/heros/${tempHero._id}`)
+      .send({name: 1234})
+      .catch(res => {
+        expect(res.status).toEqual(400);
+        expect(res.body._id).toEqual(tempHero._id);
+      });
+    });
+    it('should respond with 400', () => {
+      return superagent.put(`${API_URL}/api/heros/1234`)
+      .catch(res => {
+        expect(res.status).toEqual(404);
+      });
+    });
+  });
+  describe('testing DELETE /api/heros/:id', () => {
+    afterEach(() => Hero.remove({}));
+    beforeEach(() => {
+      return new Hero({
+        name: 'A Guy',
+        origin: 'The Couch',
+        goodOrEvil: 'Evil',
+      })
+      .save()
+      .then(hero => {
+        tempHero = hero;
+      });
+    });
+    it('should delete a hero', () => {
+      superagent.delete(`${API_URL}/api/heros/${tempHero._id}`)
+      .then(res => {
+        expect(res.status).toEqual(204);
+      });
+    });
+    it('should return 204', () => {
```

Comment:
This test is also a false positive. It is returning 404 not 204. Perhaps thats what it should return? But it isn't what we are testing.